pets/peer-finder/peer-finder.go

```diff
@@ -74,17 +75,42 @@ func main() {
 	if ns == "" {
 		ns = os.Getenv("POD_NAMESPACE")
 	}
-	if *svc == "" || ns == "" || (*onChange == "" && *onStart == "") {
-		log.Fatalf("Incomplete args, require -on-change and/or -on-start, -service and -ns or an env var for POD_NAMESPACE.")
-	}
-
 	hostname, err := os.Hostname()
 	if err != nil {
 		log.Fatalf("Failed to get hostname: %s", err)
 	}
+	var domainName string
+
+	// If domain is not provided, try to get it from resolv.conf
+	if (*domain == "") || (ns == "") {
+		resolvConfBytes, err := ioutil.ReadFile("/etc/resolv.conf")
+		resolvConf := string(resolvConfBytes)
+		if err != nil {
+			log.Fatal("Unable to read /etc/resolv.conf")
+		}
+		if ns != "" {
+			tempResolv := strings.Split(resolvConf, ("search " + ns + "."))
```

Comment:
I think "deducing cluster domain from resolv.conf when domain not set" makes sense --- as I don't have better ideas. But also be aware of that resolv.conf will not have cluster domain set up if `dnsPolicy` is set to `Default` ([example](https://github.com/kubernetes/kubernetes/blob/v1.6.6/cluster/addons/dns/kubedns-controller.yaml.base#L171)).